### DISCUSSION TITLE
Execute golang function in pb hooks

### DISCUSSION THREAD

**Discussion by starredev** [pocketbase/pocketbase#4796] [Q&A]
I was wondering if it is possible to execute a function in pb_hooks which I created in pocketbase go. I have added ldap-go into my pocketbase golang file which return the user object, From that point I want to continue with PB Hooks as my go skills are not very good at the moment..

I could do it like this way but it feels unclear

```
routerAdd("GET", "/authentication", (c) => {

    const res = $http.send({
        url: "http://localhost:8090/ldap-authentication",
        method: "POST",
        body: JSON.stringify({
            username: 'xxxx',
            password: 'xxxx'
        }),
    });
	return c.json(200, { "message": res })
})
```

**Comment by ganigeorgiev** [comment]
You can register a Go function with the [`jsvm.Config.OnInit`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.9/plugins/jsvm#Config) field but it will depend how actually your function works and what arguments it accept (_goja can convert simple values like int, string, map, etc. but if  your function expect struct as argument you'll have to create a binding for that too or a wrapper that translates to the struct_):

```go
jsvm.MustRegister(app, jsvm.Config{
    OnInit: func(vm *goja.Runtime) {
        // register custom bindings
        // (for more details see https://pkg.go.dev/github.com/dop251/goja#hdr-Functions)
        vm.Set("customFunc", yourGoFunc)
    }
})
```

**Comment by starredev** [comment]
Thanks that works :)
